src/Config/ConfigMerger.php

```diff
@@ -51,14 +68,19 @@ private function mergeConfigEnvironments(Config $config, Config $override): arra
                 continue;
             }
 
-            $environments[$name] = $this
-                ->mergeEnvironments($config->getEnvironments()[$name], $override->getEnvironments()[$name]);
+            if ($asImport) {
```

Comment:
Just split this methode, instead of a bool parameter.

Reply:
Good call!